sdk/trace/batch_span_processor.go

```diff
@@ -250,7 +250,7 @@ func (bsp *batchSpanProcessor) exportSpans(ctx context.Context) error {
 	}
 
 	if l := len(bsp.batch); l > 0 {
-		global.Debug("exporting spans", "count", len(bsp.batch), "dropped", bsp.dropped)
+		global.Debug("exporting spans", "count", len(bsp.batch), "dropped", atomic.LoadUint32(&bsp.dropped))
```

Comment:
It looks like `bsp.dropped` is never reset. Think there is a need to refactor the synchronization logic to ensure an atomic view into the whole batch being exported and then uniform reset. Until then, do we want to just remove logging this to fix the race condition?

Reply:
Not sure, my fix does not change any logic, it simply fixes the race. If you want more fixes, I don't have time to do, so up to @open-telemetry/go-maintainers to decide what to do.

Reply:
@MrAlias When would it reset?  This is just a count of how many spans have been dropped, which is independent of the success or failure of subsequent batch exports.

If we did reset on export we wouldn't be getting a useful number because drops happen when the queue is full, which is most likely when the batch is locked during export.